### PR TITLE
Allow to disable colorization

### PR DESCRIPTION
**Pull request by theo-lubert**
Most color librairies allow this in other languages
It is especially helpful in cases where colors are not supported, like say... in XCode debugger

### CODE REVIEW

**Comment by mtynior** [comment]
Hi @theo-lubert, this is great idea and I like what you did. The only thing that concerns me, is relying on the CommandLine arguments. I think it would be better to extend the `String` class with a property - let say `static var isColorizationEnabled: Bool = true` - and allow users to decide themselves how they want to disable colorization.

**Comment by theo-lubert** [comment]
Hi @mtynior , relying on CommandLine is fine since it is part of the Swift Standard Library.
But that's not really your point, we should be able to disable color through String.enableColor too.
So I did that also, see my last commit.

I kept --no-color though, and I gave it a higher priority. Since that was kind of the point of the pull request.
In my opinion, a developper should not be able to enable/force color if the user (of a cmdline tool for example) want to disable it with --no-color, most likely for a good reason (xcode, log, ssh, etc)

**Comment by theo-lubert** [comment]
Now I also get that is a bit opinionated, and it might not be the job of this library to support a command line option... but if we don't, I feel like most developers simply won't support the --no-color option at all
I believe it's worth it. Your call

**Comment by mtynior** [comment]
Supporting the `--no-color` flag out of the box solves the problem you had, but might cause issues to other developers. Let's say that Jonh created an image processing tool that uses the ColorizeSwift to prettify the output. In the newest version, John added filter that is triggered by `--no-color` flag. The filter worked fine but Jonh also lost the colorization of the terminal output. 
Jonh could change name of the flag in the tool. But it doesn't feel right to force devs to change their designs just because library tried to "outsmart" the developer.
Adding the flag that allows to disable the colorization solves the fundamental problem you had. How the flag is set should be up to developer who uses the library. For you, it is command line argument, but for someone else it might be the config file. Also someone might not want to allow to disable colorization or it might want to use different flag.
I hope, that now, you can understand why supporting the `--no-color` flag out of the box might cause more harm than good.

File: Source/ColorizeSwift.swift [hunk 1]
```diff
@@ -59,6 +61,11 @@ public struct TerminalStyle {
 }
 
 extension String {
+
+    public static var enableColor = true
```
Comment (mtynior) [inline Source/ColorizeSwift.swift]: The `enableColor` sounds like an action.With that naming convention it should be a function not the property. We want to represent the state. It should be noun not a verb. As it is a boolean it should also be prefixed with "is" e.g. `isColorizationEnabled`.

**Comment by theo-lubert** [comment]
You're right.
I changed the naming, and made `--no-color` optional/customizable so developers can opt out as they see fit.
I still feel like it can do some good though, and should be left enabled by default, but I would understand if you don't agree. Let me know if you do, and I'll disable it.

Aside from that, the feature should be good to go.

**Comment by mtynior** [comment]
I like that you solved the problems I mentioned before, but as I said before, goal of this library is to provide minimalistic api to the problem of terminal's output colorization.
I can agree that disabling colorization should be core feature, but doing it from command line is too much. Especially when it can be done by the developers themselves with only one line of code: `String.isColorizationEnabled = CommandLine.arguments.contains("--no-color")`. 
This approach simplifies library's code and makes usage more explicit - more on that below.

That being said, I can accept pull request that provides ability to disable colorization via single boolean property, but I cannot merge code that operates on command line arguments.

Now, from technical point of view. If you disable colorization only via command line, the `isColorizationEnabled` will return wrong value. 
Beside that, current approach will be evaluating the presence of the "--no-color" flag for every single time when `applyStyle` is called, which is not optimal. Having only one property `isColorizationEnabled` solves this issue, because `String.isColorizationEnabled = CommandLine.arguments.contains("--no-color")` will be evaluated only once.

**Comment by theo-lubert** [comment]
Alright, I removed it, all that is left is the boolean.

For the record, the only concern I have is that a developer could mistakenly re-enable colorization even though the cmd line user disabled it. That could lead to some issues, but this can be solve elsewhere.

Also, regarding the technical concerns you had:
- isColorizationEnabled would be wrong. Granted, there were ways to solve this though
- If you read the code,  CommandLine.arguments.contains("--no-color") is only evaluated once, and if you change the option flag. I made sure it would not be a performance drag.

**Pull request merged**